src/controllers/userController.js

```diff
@@ -0,0 +1,102 @@
+const userModel = require("../models/userSchema");
+const bcrypt = require("bcrypt");
+
+/******************************************************
+ * @SIGNUP
+ * @route /api/v1/signup
+ * @method POST
+ * @description singUp function for creating new user
+ * @body firstname,lastname,email, password
+ * @returns User Object
+ ******************************************************/
+
+const signUp = async (req, res) => {
+  try {
+    const userInfo = userModel(req.body);
+    const result = await userInfo.save();
+    return res.status(200).json({
+      success: true,
+      data: result,
+    });
+  } catch (error) {
+    //verify is user is already exists
+    if (error.code === 11000) {
+      return res.status(400).json({
+        success: false,
+        message: "User already exists",
+      });
+    }
+    return res.status(400).json({
+      success: false,
+      message: error.message,
+    });
+  }
+};
+/******************************************************
+ * @LOGIN
+ * @route /api/v1/login
+ * @method POST
+ * @description verify user and send cookie with jwt token
+ * @body email , password
+ * @returns User Object , cookie
+ ******************************************************/
+
+const logIn = async (req, res) => {
+  try {
+    const { email, password } = req.body;
+    const user = await userModel.findOne({ email }).select("+password");
+    if (!email || !(await bcrypt.compare(password, user.password))) {
+      return res.status(400).json({
+        success: false,
+        message: "Invalid Credentials",
+      });
+    }
+    //create token
+
+    const token = user.jwtToken();
+    user.password = undefined;
+    const cookiesOptions = {
+      maxAge: 24 * 60 * 60 * 1000,
```

Comment:
The maxAge value would work well as a constant name something like `MAX_AGE` at the top of this module. That way it'd be easy to identify this important info for anyone debugging. What do you think about this?

Reply:
I really never tried to add maxAge as a constance